source/geometry_model/ellipsoidal_chunk.cc

```diff
@@ -110,6 +113,40 @@ namespace aspect
       return phi_theta_d;
     }
 
+    template <int dim>
+    Point<3>
+    EllipsoidalChunk<dim>::EllipsoidalChunkGeometry::push_forward_topography(const Point<3> &phi_theta_d_hat) const
+    {
+      const double d_hat = phi_theta_d_hat[2]; // long, lat, depth
+      Point<dim-1> phi_theta;
+      const double rad_to_degree = 180/numbers::PI;
+      if (dim == 3)
+        phi_theta = Point<dim-1>(phi_theta_d_hat[0] * rad_to_degree,phi_theta_d_hat[1] * rad_to_degree);
+      const double h     = topography->value(phi_theta);
```

Comment:
Sure. I was not saying it is possible to do it otherwise, I just do not see from the documentation why we have to do it this way. I am not familiar with these parts of the code, and the documentation of the class ( "A class which describes the manifold.") does not help me much in understanding what it does (maybe you can expand this a bit, since you know more about what is happening here?). 
I now see that it needs to store some access to the initial_topography_model, because deal.II will call the function upon refinement from the outside (i.e. not from the EllipsoidalChunk object). However, I still do not see, why it needs to be static and public. Could we not make a private pointer that is either set by a public `EllipsoidalChunkGeometry::set_initial_topography` function called from `EllipsoidalChunk<dim>::initialize()`? Or a private pointer that is initialized in the constructor of `EllipsoidalChunkGeometry`? In the second case we would want to generate the `EllipsoidalChunkGeometry` object in the `EllipsoidalChunk::initialize` function, e.g. by storing it as a unique_ptr instead of a member object (look at the handling of `lateral_viscosity_lookup` in the `Steinberger` material model).
I am just concerned that public static members should only be used if there is absolutely no other possibility, and it is not documented why we need to do it here (and I do not see it either).